### PR TITLE
build(property-common): use fluid-tsc -- secret nested package.json

### PR DESCRIPTION
**Pull request by DLehenbauer**
[AB#7379](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7379)

The 'property-common' package contains a nested 'platform-dependent' package that is used to import browser vs. node implementations of crypto random.

It looks like this is for uuid generation.  Probably can substitute the uuid package we use elsewhere in Fluid, or a 3rd party isomorphic crypto random lib.

```
@fluid-experimental/property-common: src/guidUtils.ts:13:43 - error TS2834: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.
@fluid-experimental/property-common: 
@fluid-experimental/property-common: 13 import { generateRandomUInt32Array } from "../platform-dependent";
@fluid-experimental/property-common:                                              ~~~~~~~~~~~~~~~~~~~~~~~
@fluid-experimental/property-common: 
@fluid-experimental/property-common: 
@fluid-experimental/property-common: Found 1 error in src/guidUtils.ts:13
```

Semi-automatic PR to switch to fluid-tsc

### CODE REVIEW

File: experimental/PropertyDDS/packages/property-common/src/guidUtils.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@
  */
 
 import base64js from "base64-js";
-import { generateRandomUInt32Array } from "../platform-dependent";
+import { v4 as uuid } from "uuid";
```
Comment (jason-ha) [inline experimental/PropertyDDS/packages/property-common/src/guidUtils.ts:13]: @ChumpChief has [AB#7451](https://dev.azure.com/fluidframework/internal/_workitems/edit/7451): "Remove @fluid-experimental/platform-dependent package" (which I think is meant to say "Remove @fluid-internal/platform-dependent").
There are comments that suggest it wouldn't be this simple. Is this change viable?

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: experimental/PropertyDDS/packages/property-common/src/test/datastructures/collection.spec.ts [hunk 2]
```diff
@@ -6,10 +6,11 @@
 /**
  * @fileoverview In this file, we will test the functions exported by datastructres/collection.js
  */
-
+import lodash from "lodash";
+// eslint-disable-next-line @typescript-eslint/unbound-method -- 'lodash' import workaround.
```
[the remark this answers is not in the document]
Comment (RishhiB) [reply experimental/PropertyDDS/packages/property-common/src/test/datastructures/collection.spec.ts]: removed, it was unused

File: experimental/PropertyDDS/packages/property-common/src/error_objects/httpErrorNoStack.ts [hunk 3]
```diff
@@ -3,8 +3,12 @@
...
  */
 
-import { FlaggedError } from "./flaggedError";
-import { HTTPError } from "./httpError";
+import { FlaggedError } from "./flaggedError.js";
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { HTTPError } from "./httpError.js";
```
Comment (tylerbutler) [inline experimental/PropertyDDS/packages/property-common/src/error_objects/httpErrorNoStack.ts:11]:
```suggestion
import { HTTPError } from "./httpError.js";
```

Comment (RishhiB) [reply experimental/PropertyDDS/packages/property-common/src/error_objects/httpErrorNoStack.ts:11]: before adding this line into the code, it was failing `npm run checks`

**Pull request closed without merging**